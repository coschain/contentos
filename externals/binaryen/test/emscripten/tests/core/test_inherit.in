#include <stdio.h>
struct Parent {
  int x1, x2;
};
struct Child : Parent {
  int y;
};
int main() {
  Parent a;
  a.x1 = 50;
  a.x2 = 87;
  Child b;
  b.x1 = 78;
  b.x2 = 550;
  b.y = 101;
  Child* c = (Child*)&a;
  c->x1++;
  c = &b;
  c->y--;
  printf("*%d,%d,%d,%d,%d,%d,%d*\n", a.x1, a.x2, b.x1, b.x2, b.y, c->x1, c->x2);
  return 0;
}
