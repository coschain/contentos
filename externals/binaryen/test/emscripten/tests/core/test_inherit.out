*51,87,78,550,100,78,550*